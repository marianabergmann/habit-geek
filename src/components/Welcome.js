import React, { useState } from "react";
import { navigate } from "@reach/router";
import "./Welcome.scss";
import avatar1 from "./../imgs/avatars/001-burglar.svg";
import avatar2 from "./../imgs/avatars/002-woman.svg";
import avatar3 from "./../imgs/avatars/003-superhero.svg";
import avatar4 from "./../imgs/avatars/004-robot.svg";
import avatar5 from "./../imgs/avatars/005-dragon.svg";
import avatar6 from "./../imgs/avatars/006-cyborg.svg";

const Welcome = ({ registerUser }) => {
  const [name, setName] = useState("");
  const [selectedAvatar, setAvatar] = useState(avatar1);

  const register = () => {
    registerUser(name, selectedAvatar);
    navigate("/dashboard");
  };

  return (
    <div>
      <h2>Welcome!</h2>
      <label htmlFor="Name">
        Name:
        <input
          id="Name"
          value={name}
          onChange={e => setName(e.target.value)}
        ></input>
      </label>
      <div>
        Avatar:
        {[avatar1, avatar2, avatar3, avatar4, avatar5, avatar6].map(
          (avatar, index) => (
            <input
              key={`avatar-${index}`}
              onClick={() => setAvatar(avatar)}
              type="image"
              src={avatar}
              alt="Search"
              className="welcome__avatar"
            ></input>
          )
        )}
      </div>

      <button onClick={register}>Done!</button>
    </div>
  );
};

export default Welcome;
