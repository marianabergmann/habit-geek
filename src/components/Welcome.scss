.welcome__avatar {
  width: 30px;
  border-radius: 50%;
}
