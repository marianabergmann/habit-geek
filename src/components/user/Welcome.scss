.welcome {
  width: 100%;
  max-width: 340px;
  padding: 15px;
  margin: auto;
}

.welcome__title {
  text-align: center;
}

.welcome__avatar-container {
  display: flex;
  justify-content: space-around;
  height: 50px;
}

.welcome__avatar {
  width: 40px;
  border-radius: 50%;
}

.welcome__avatar--selected {
  transition-property: width;
  transition-duration: 0.2s;
  border: 3px solid var(--primary);
  outline: none;
  width: 50px;
}
