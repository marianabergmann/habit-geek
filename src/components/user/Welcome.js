import React, { useState } from "react";
import { navigate } from "@reach/router";
import "./Welcome.scss";
import avatar1 from "./../../imgs/avatars/001-burglar.svg";
import avatar2 from "./../../imgs/avatars/002-woman.svg";
import avatar3 from "./../../imgs/avatars/003-superhero.svg";
import avatar4 from "./../../imgs/avatars/004-robot.svg";
import avatar5 from "./../../imgs/avatars/005-dragon.svg";
import avatar6 from "./../../imgs/avatars/006-cyborg.svg";

const avatars = [avatar1, avatar2, avatar3, avatar4, avatar5, avatar6];

const Welcome = ({ name: initialName, avatar, registerUser }) => {
  const [name, setName] = useState(initialName);
  const [selectedAvatar, setAvatar] = useState(avatar ? avatar : "");

  const register = () => {
    registerUser(name, selectedAvatar);
    navigate("/habits");
  };

  return (
    <div className="welcome">
      <h2 className="welcome__title">Welcome!</h2>
      <div className="mb-3">
        <label htmlFor="welcome-name">Name:</label>
        <input
          id="welcome-name"
          className="form-control"
          value={name}
          onChange={e => setName(e.target.value)}
        ></input>
      </div>
      <div className="mb-3">
        <label htmlFor="welcome-avatar">Avatar:</label>
        <div className="input-group welcome__avatar-container">
          {avatars.map((avatar, index) => (
            <input
              key={`avatar-${index}`}
              onClick={() => setAvatar(avatar)}
              type="image"
              src={avatar}
              alt="Search"
              className={`welcome__avatar ${
                avatar === selectedAvatar ? "welcome__avatar--selected" : ""
              }`}
            ></input>
          ))}
        </div>
      </div>
      <hr className="mb-4" />
      <button className="btn btn-primary btn-lg btn-block" onClick={register}>
        Done!
      </button>
    </div>
  );
};

export default Welcome;
