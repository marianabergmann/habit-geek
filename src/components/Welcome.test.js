import React from "react";
import { navigate } from "@reach/router";

import { shallow } from "enzyme";
import Welcome from "./Welcome";

jest.mock("@reach/router", () => ({
  navigate: jest.fn()
}));

afterAll(() => {
  jest.unmock("@reach/router");
});

describe("Welcome component", () => {
  it("Renders static elements", () => {
    const welcome = shallow(<Welcome />);
    expect(welcome).toMatchSnapshot();
  });

  describe("on register", () => {
    it("Register and navigates to Dashboard", () => {
      const registerUserAction = jest.fn();
      const welcome = shallow(<Welcome registerUser={registerUserAction} />);
      welcome.find("#Name").simulate("change", { target: { value: "Rafa" } });
      welcome
        .find(".welcome__avatar")
        .at(0)
        .simulate("click");
      welcome.find("button").simulate("click");

      expect(registerUserAction).toHaveBeenCalled();
      expect(registerUserAction).toHaveBeenCalledWith(
        "Rafa",
        "001-burglar.svg"
      );
      expect(navigate).toHaveBeenCalledWith("/dashboard");
    });
  });
});
