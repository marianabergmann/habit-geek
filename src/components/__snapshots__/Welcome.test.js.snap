// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Welcome 1`] = `
<div>
  <h2>
    Welcome!
  </h2>
  <label
    htmlFor="Name"
  >
    Name:
    <input
      id="Name"
      onChange={[Function]}
      value=""
    />
  </label>
  <div>
    Avatar:
    <img
      alt="Avatar 1"
      className="avatar"
      src="001-burglar.svg"
    />
    <img
      alt="Avatar 1"
      className="avatar"
      src="002-woman.svg"
    />
    <img
      alt="Avatar 1"
      className="avatar"
      src="003-superhero.svg"
    />
    <img
      alt="Avatar 1"
      className="avatar"
      src="004-robot.svg"
    />
    <img
      alt="Avatar 1"
      className="avatar"
      src="005-dragon.svg"
    />
    <img
      alt="Avatar 1"
      className="avatar"
      src="006-cyborg.svg"
    />
  </div>
  <button>
    Done!
  </button>
</div>
`;
